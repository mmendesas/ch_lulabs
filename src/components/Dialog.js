import React from "react";

const Dialog = ({ visible, onCloseClick, children }) => (
  <section className={visible ? "dialog dialog--active" : "dialog"}>
    <div className="close-thik" onClick={onCloseClick} />
    {children}
  </section>
);

export default Dialog;
