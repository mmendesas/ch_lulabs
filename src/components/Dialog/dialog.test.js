import React from 'react';
import { render } from 'react-testing-library';
import Dialog from './index';

describe('[Component] - Dialog', () => {
  it('should be match snapshot', () => {
    const { getByText } = render(<Dialog>my dialog</Dialog>);
    expect(getByText('my dialog')).toMatchSnapshot();
  });

  it('renders dialog with default values', () => {
    const { getByText } = render(<Dialog>my dialog</Dialog>);
    expect(getByText('my dialog').className).toEqual('dialog');
  });

  it('renders dialog with default values', () => {
    const { getByText } = render(<Dialog visible>my dialog</Dialog>);
    expect(getByText('my dialog').className).toEqual('dialog dialog--active');
  });
});
