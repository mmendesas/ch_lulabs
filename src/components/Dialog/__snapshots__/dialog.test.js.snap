// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - Dialog should be match snapshot 1`] = `
<section
  class="dialog"
>
  <div
    class="close-thik"
  />
  my dialog
</section>
`;
